src/coreclr/jit/assertionprop.cpp

```diff
@@ -3209,8 +3214,12 @@ GenTree* Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTree* tree)
                 if ((unsigned_abs(conValTree->AsIntCon()->IconValue()) <= 0xFFF) ||
                     isPow2(conValTree->AsIntCon()->IconValue()))
                 {
+                    // Small integers are likely containable on arm
                     keepPropagating = true;
                 }
+#elif TARGET_XARCH
+                // All integer constants are cheap to materialize on xarch
+                keepPropagating = true;
```

Comment:
Is this right?

Those that fit in 8-bits (generally sign-extended, but sometimes zero-extended) are often (but not always) containable, but if its `2-`, `4-`, or 8-bytes` then it generally needs a separate instruction and is more expensive.

Reply:
@tannergooding I agree but in this PR I only try to avoid memory loads - there are too many pitfalls this PR can't take into account like register-pressure, cached constants which live across calls on ABIs without callee-saved regs, etc - it's just a quick heuristic at this point

Reply:
It might be worth logging an issue for someone to experiment here a bit.

I'd expect that most cases of explicit `mov reg, imm` are better handled by hoisting as it impacts the size of the loop and doesn't allow optimizations in many scenarios.

Reply:
Sure thing, I am actually going to spend some time on it after this PR because it's even more important for ARM where it might ends up with 4 mov per single constant